pkg/server/cloud/aws/instances.go

```diff
@@ -455,6 +496,12 @@ func (e *AwsEC2) StartDedicatedNode(node *api.Node, image cloud.Image, metadata,
 	volSizeGiB := cloud.ToSaneVolumeSize(node.Spec.Resources.VolumeSize, image)
 	devices := e.getBlockDeviceMapping(image, volSizeGiB)
 	networkSpec := e.getInstanceNetworkSpec(node.Spec.Resources.PrivateIPOnly)
+	ctx, _ := context.WithTimeout(context.Background(), 30 * time.Second)
```

Comment:
The second return value should be called before returning this function to prevent a resource leak.

Reply:
:+1: added explicit `cancel()` before sending `RunInstance` request